enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,549 @@
+
+1. User sits down at their computer.
+2. The user creates their `install-config.yaml`, including extra
+   values in the `bootstrapInPlace` section (since that enables
+   deploying a single-node).
+   * `managementCPUIDs` is a CPU set specifier for the CPUs to add to
+     the isolated set used for management components. The default is
+     empty, and when the value is empty the management workload
+     partitioning feature is disabled. The rest of the steps below
+     assume it is enabled, unless otherwise stated.
+3. The user runs the installer.
+4. The installer uses `managementCPUIDs` to generate an extra machine
+   config manifest to configure CRI-O to process management workloads
+   in a special way.
+5. The installer creates a machine config manifest to write a
+   configuration file for kubelet. The file should only be readable by
+   the kubelet.
+6. The installer updates the status fields of the Infrastructure
+   config resource to enable the feature for the admission hook.
+7. The kubelet starts up and finds the configuration file enabling the
+   new feature.
+8. The kubelet reads static pod definitions. It replaces the `cpu`
+   requests with `management-cpu` requests of the same value and adds
+   an annotation for CRI-O with the same value.
+9. Something schedules a regular pod with the
+   `io.openshift.management: true` label in a namespace with the
+   `io.openshift.management: true` label.
+10. The admission hook modifies the pod, replacing the CPU requests
+    with management CPU requests and adding an annotation for CRI-O.
+11. The scheduler sees the new pod and finds available management CPU
+    resources on the node. The scheduler places the pod on the node.
+12. Repeat steps 9-11 until all pods are running.
+13. Single-node deployment comes up with management components
+    constrained to subset of available CPUs
+
+#### CRI-O Changes
+
+CRI-O will be updated to support a new configuration value that a user
+can specify, `mgmt_ctr_cpuset`.
+
+```ini
+[crio.runtime]
+mgmt_ctr_cpuset = “0-1”
+```
+
+This field describes the CPU set that management workloads will be
+configured to use.
+
+CRI-O will be configured to support a new annotation on pods,
+`io.openshift.management.cores`.
+
+```ini
+[crio.runtime.runtimes.runc]
+  runtime_path = "/usr/bin/runc"
+  runtime_type = "oci"
+  allowed_annotations = ["io.openshift.management.cores"]
+```
+
+Pods that have the `io.openshift.management.cores` annotation will
+have their cpuset configured to the value in `mgmt_ctr_cpuset`, as
+well as have their CPU shares configured to the value of the
+annotation.
+
+Note that this field does not conflict with the `infra_ctr_cpuset`
+config option, as the infra container will still be put in that
+cpuset.  They can be configured as the same value if the infra
+container should also be considered to be managed.
+
+#### API Server Admission Hook
+
+A new admission hook in the kubernetes API server within OpenShift
```

Comment:
I'm curious why we prefer admission hook vs being explicit for specific set of pods?